### PR TITLE
distro: Refactor exporterConfig

### PR DESCRIPTION
**Pull request by pellared**
### Proposed changes

- Remove the OTLP exporter env var handling as OTel Go SDK handles it already in the same way (has the same default)
- Remove `UseTLS` as it was not necessary.

### CODE REVIEW

File: distro/exporter.go [hunk 1]
```diff
@@ -41,24 +41,11 @@ var exporters = map[string]traceExporterFunc{
...
-	if c.Endpoint == "" {
-		if endpoint := func() string {
-			// Allow the exporter to use environment variables if set.
-			if _, ok := os.LookupEnv(otelExporterOTLPEndpointKey); ok {
-				return ""
-			}
-			if _, ok := os.LookupEnv(otelExporterOTLPTracesEndpointKey); ok {
-				return ""
-			}
-			return defaultOTLPEndpoint
```
Comment (MrAlias) [inline distro/exporter.go:53]: Removing this is not going to work. If we do not check this and use our default value but the user set either environment variable, those environment variables will be ignored. This comes from the fact that OTel-Go will override environment variables with any passed argument.

**Review by MrAlias — CHANGES_REQUESTED** [review]

File: distro/exporter.go (hunk 1 above)
Comment (pellared) [reply distro/exporter.go:53]: > OTel-Go will override environment variables with any passed argument.

Is it not considered a bug? _Or maybe I understand something bad._ I think no other SDK works like that. Also, I would not come up with this after reading https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/otlp/otlptrace#environment-variables

Comment (MrAlias) [reply distro/exporter.go:53]: > Is it not considered a bug?

Not by all. I agree it is not idea, but it has become the released behavior.

> Also, I would not come up with this after reading

Why not? What do you take away from this?

> [Configuration using options have precedence over the environment variables.](https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/otlp/otlptrace#environment-variables)

For completeness:

https://github.com/open-telemetry/opentelemetry-go/blob/a4ea45c38cbcf76c43ab0555b2d4bcacbbfdce9a/exporters/otlp/otlptrace/internal/otlpconfig/options.go#L101-L104

Comment (pellared) [reply distro/exporter.go:53]: I will take another look at it tomorrow. Right now, I do not see a problem in my proposed change...

An example scenario when my proposed change would not work will hopefully enlighten me. Then I could debug it and understand it 😄

Comment (MrAlias) [reply distro/exporter.go:53]: Is it?! I thought the Jaeger endpoint was different. Let me check.

Comment (MrAlias) [reply distro/exporter.go:53]: err, yeah, this is for the OTLP exporter. Yeah, you are correct that is the same.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**